src/canvas/drawing_utils.rs

```diff
@@ -1,149 +1,35 @@
-        "soft width min length != soft width max length!"
-    );
-    debug_assert!(
-        soft_widths_max.len() == soft_widths_desired.len(),
-        "soft width max length != soft width desired length!"
-    );
-
-    if total_width > 2 {
-        let initial_width = total_width - 2;
-        let mut total_width_left = initial_width;
-        let mut column_widths: Vec<u16> = vec![0; hard_widths.len()];
-        let range: Vec<usize> = if left_to_right {
-            (0..hard_widths.len()).collect()
-        } else {
-            (0..hard_widths.len()).rev().collect()
-        };
-
-        for itx in &range {
-            if let Some(Some(hard_width)) = hard_widths.get(*itx) {
-                // Hard width...
-                let space_taken = min(*hard_width, total_width_left);
-
-                // TODO [COLUMN MOVEMENT]: Remove this
-                if *hard_width > space_taken {
-                    break;
-                }
-
-                column_widths[*itx] = space_taken;
-                total_width_left -= space_taken;
-                total_width_left = total_width_left.saturating_sub(1);
-            } else if let (
-                Some(Some(soft_width_max)),
-                Some(Some(soft_width_min)),
-                Some(Some(soft_width_desired)),
-            ) = (
-                soft_widths_max.get(*itx),
-                soft_widths_min.get(*itx),
-                soft_widths_desired.get(*itx),
-            ) {
-                // Soft width...
-                let soft_limit = max(
-                    if soft_width_max.is_sign_negative() {
-                        *soft_width_desired
-                    } else {
-                        (*soft_width_max * initial_width as f64).ceil() as u16
-                    },
-                    *soft_width_min,
-                );
-                let space_taken = min(min(soft_limit, *soft_width_desired), total_width_left);
-
-                // TODO [COLUMN MOVEMENT]: Remove this
-                if *soft_width_min > space_taken {
-                    break;
-                }
-
-                column_widths[*itx] = space_taken;
-                total_width_left -= space_taken;
-                total_width_left = total_width_left.saturating_sub(1);
-            }
-        }
-
-        while let Some(0) = column_widths.last() {
-            column_widths.pop();
-        }
-
-        if !column_widths.is_empty() {
-            // Redistribute remaining.
-            let amount_per_slot = total_width_left / column_widths.len() as u16;
-            total_width_left %= column_widths.len() as u16;
-            for (index, width) in column_widths.iter_mut().enumerate() {
-                if index < total_width_left.into() {
-                    *width += amount_per_slot + 1;
-                } else {
-                    *width += amount_per_slot;
-                }
-            }
-        }
-
-        column_widths
-    } else {
-        vec![]
-    }
-}
```

Comment:
Deleted since this responsibility was moved to `TableState`, and this PR means nothing relies on it anymore.